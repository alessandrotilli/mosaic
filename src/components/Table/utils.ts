export const COLUMN_CHECKBOX_PATH = "checkbox-selection";
export const COLUMN_ROW_ACTIONS_PATH = "row-actions";

export const TOOLBAR_HEIGHT = 64;
export const TOOLBAR_HEIGHT_MOBILE = 56;

export const PAGINATION_TOOLBAR_HEIGHT = 64;
