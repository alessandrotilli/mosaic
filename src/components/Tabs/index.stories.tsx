import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";

import { localeDecorator } from "../../utils/mocks/LocaleMock";
import getDocsPage from "../../utils/stories";
import Progress from "../Progress";

import Tabs, { DATA_CY_DEFAULT } from ".";

const COMPONENT_NAME = "Tabs";
Tabs.displayName = COMPONENT_NAME;

export default {
  title: "Navigation/Tabs",
  component: Tabs,
  decorators: [localeDecorator],
  parameters: {
    docs: {
      ...getDocsPage({
        basedOn: {
          label: "MUI Tabs Component",
          url: "https://mui.com/material-ui/react-progress/",
        },
        component: COMPONENT_NAME,
        e2eTestInfo: {
          dataCyDefault: DATA_CY_DEFAULT,
        },
      }),
    },
  },
} as ComponentMeta<typeof Tabs>;

const Template: ComponentStory<typeof Tabs> = (args) => <Tabs {...args} dataCy={DATA_CY_DEFAULT} />;

export const Primary = Template.bind({});
Primary.args = {
  labelList: [{ label: "PAGE1" }, { label: "PAGE2" }, { label: "PAGE3" }],
};

export const Color = Template.bind({});
Color.args = { ...Primary.args, color: "secondary" };

export const Custom = Template.bind({});
Custom.args = {
  labelList: [
    { label: "PAGE1" },
    { label: "PAGE2" },
    { label: "PAGE3", children: <Progress type="Circular" withLabel={false} /> },
  ],
};

export const Orientation = Template.bind({});
Orientation.args = { ...Custom.args, orientation: "vertical" };
