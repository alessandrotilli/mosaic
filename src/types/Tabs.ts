import { IBase } from "./Base";
import { ButtonColor } from "./Button";

export type TabType = {
  label: string;
  wrapped?: boolean;
  content?: React.ReactNode;
  disabled?: boolean;
  textColor?: ButtonColor;
};

export type TabsType = IBase & {
  tabList: TabType[];
  color?: ButtonColor;
  orientation?: "horizontal" | "vertical";
};
