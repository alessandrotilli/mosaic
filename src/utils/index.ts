import { ChangeEvent, MouseEvent } from "react";

export interface ISubpart {
  label: string;
  value?: Function;
}

export interface ISubpartSuffix {
  label: string;
  suffix: string;
}

export type ISubpartMap<T extends string = string> = {
  [key in T]: ISubpart;
};

export const DATA_CY_SUFFIX_SEPARATOR = "-";

/**
 * Gets composed dataCy for components with subparts
 * @param dataCy the dataCy prefix
 * @param subpart the subpart to consider
 * @param args optional arguments for subpart evaluation
 */
export const getComposedDataCy = (dataCy: string, subpart: ISubpart, args?: any): string => {
  const { label, value } = subpart;
  return `${dataCy}${DATA_CY_SUFFIX_SEPARATOR}${!value ? slugify(label) : value(args)}`;
};

/**
 * Gets all composed dataCy for all listed subparts
 * @param subpartMap the map of all subparts to consider
 */
export const getAllComposedDataCy = (subpartMap: ISubpartMap): ISubpartSuffix[] =>
  Object.values(subpartMap).map(({ label, value }) => ({
    label,
    suffix: !value ? slugify(label) : value(),
  }));

/**
 * Gets an object property by path, if path is nested is executed recursively
 * @param object the source object
 * @param path the desired path
 */
export const getObjectProperty = (object: any, path: string): any => {
  if (!object) {
    return undefined;
  }

  if (!path.includes(".")) {
    return object[path];
  }

  const pathParts = path.split(".");
  const basePath = pathParts[0];
  if (!object[basePath]) {
    return undefined;
  }

  const baseObject = object[basePath];
  return getObjectProperty(baseObject, path.replace(`${basePath}.`, ""));
};

/**
 * Creates slug or kebab-case string from value
 * @param value the value to slugify
 */
export const slugify = (value: string) =>
  value
    .toLowerCase()
    .replace(/\s+/g, "-")
    .replace(/[^\w-]+/g, "");

/**
 * Suppresses default behaviour and propagation of a given event
 * @param event the event to suppress
 */
export const suppressEvent = (event: ChangeEvent<any> | MouseEvent<any, MouseEvent> | null) => {
  if (!event) {
    return;
  }

  event.preventDefault();
  event.stopPropagation();
};
