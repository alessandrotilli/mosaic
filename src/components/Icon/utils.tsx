import React from "react";
import MUIAccountCircleIcon from "@material-ui/icons/AccountCircle";
import MUIAddIcon from "@material-ui/icons/Add";
import MUIAppsIcon from "@material-ui/icons/Apps";
import MUIArrowUpwardIcon from "@material-ui/icons/ArrowUpward";
import MUIBlockIcon from "@material-ui/icons/Block";
import MUIBrightness4Icon from "@material-ui/icons/Brightness4";
import MUIBusinessIcon from "@material-ui/icons/Business";
import MUICheckIcon from "@material-ui/icons/Check";
import MUICheckBoxIcon from "@material-ui/icons/CheckBox";
import MUICheckBoxOutlineBlankIcon from "@material-ui/icons/CheckBoxOutlineBlank";
import MUIChevronLeftIcon from "@material-ui/icons/ChevronLeft";
import MUIChevronRightIcon from "@material-ui/icons/ChevronRight";
import MUICloseIcon from "@material-ui/icons/Close";
import MUICropIcon from "@material-ui/icons/Crop";
import MUIDeleteIcon from "@material-ui/icons/Delete";
import MUIDetailsIcon from "@material-ui/icons/Details";
import MUIDraftsIcon from "@material-ui/icons/Drafts";
import MUIEditIcon from "@material-ui/icons/Edit";
import MUIErrorIcon from "@material-ui/icons/Error";
import MUIFilterListIcon from "@material-ui/icons/FilterList";
import MUIFirstPageIcon from "@material-ui/icons/FirstPage";
import MUIGroupWorkIcon from "@material-ui/icons/GroupWork";
import MUIHomeIcon from "@material-ui/icons/Home";
import MUIKeyboardArrowDownIcon from "@material-ui/icons/KeyboardArrowDown";
import MUIKeyboardArrowUpIcon from "@material-ui/icons/KeyboardArrowUp";
import MUILanguageIcon from "@material-ui/icons/Language";
import MUILastPageIcon from "@material-ui/icons/LastPage";
import MUIListIcon from "@material-ui/icons/List";
import MUIMailIcon from "@material-ui/icons/Mail";
import MUIMenuIcon from "@material-ui/icons/Menu";
import MUINotificationsIcon from "@material-ui/icons/Notifications";
import MUIOpenInNew from "@material-ui/icons/OpenInNew";
import MUIPauseIcon from "@material-ui/icons/Pause";
import MUIPlayArrowIcon from "@material-ui/icons/PlayArrow";
import MUIPlaylistAddCheckIcon from "@material-ui/icons/PlaylistAddCheck";
import MUIPowerSettingsNewIcon from "@material-ui/icons/PowerSettingsNew";
import MUIPrintIcon from "@material-ui/icons/Print";
import MUIRefreshIcon from "@material-ui/icons/Refresh";
import MUISaveIcon from "@material-ui/icons/Save";
import MUIScheduleIcon from "@material-ui/icons/Schedule";
import MUISearchIcon from "@material-ui/icons/Search";
import MUISendIcon from "@material-ui/icons/Send";
import MUISettingsIcon from "@material-ui/icons/Settings";
import MUISignalWifi1BarOutlinedIcon from "@material-ui/icons/SignalWifi1BarOutlined";
import MUITimelineIcon from "@material-ui/icons/Timeline";
import MUIVisibilityIcon from "@material-ui/icons/Visibility";
import MUIVisibilityOffIcon from "@material-ui/icons/VisibilityOff";
import MUIWarningIcon from "@material-ui/icons/Warning";
import MUIWifiOffOutlinedIcon from "@material-ui/icons/WifiOffOutlined";
import MUIWifiOutlinedIcon from "@material-ui/icons/WifiOutlined";

import { Icons, IIconsCatalog } from "../../types/Icon";

export const iconsCatalog: IIconsCatalog = {
  [Icons.account]: <MUIAccountCircleIcon />,
  [Icons.add]: <MUIAddIcon />,
  [Icons.apps]: <MUIAppsIcon />,
  [Icons.block]: <MUIBlockIcon />,
  [Icons.brightness]: <MUIBrightness4Icon />,
  [Icons.business]: <MUIBusinessIcon />,
  [Icons.chart_timeline]: <MUITimelineIcon />,
  [Icons.check]: <MUICheckIcon />,
  [Icons.checkbox]: <MUICheckBoxIcon />,
  [Icons.checkbox_empty]: <MUICheckBoxOutlineBlankIcon />,
  [Icons.clock]: <MUIScheduleIcon />,
  [Icons.close]: <MUICloseIcon />,
  [Icons.connection_slow]: <MUISignalWifi1BarOutlinedIcon />,
  [Icons.crop]: <MUICropIcon />,
  [Icons.delete]: <MUIDeleteIcon />,
  [Icons.details]: <MUIDetailsIcon />,
  [Icons.down]: <MUIKeyboardArrowDownIcon />,
  [Icons.edit]: <MUIEditIcon />,
  [Icons.error]: <MUIErrorIcon />,
  [Icons.filter]: <MUIFilterListIcon />,
  [Icons.first]: <MUIFirstPageIcon />,
  [Icons.group]: <MUIGroupWorkIcon />,
  [Icons.home]: <MUIHomeIcon />,
  [Icons.language]: <MUILanguageIcon />,
  [Icons.last]: <MUILastPageIcon />,
  [Icons.left]: <MUIChevronLeftIcon />,
  [Icons.list_check]: <MUIPlaylistAddCheckIcon />,
  [Icons.list]: <MUIListIcon />,
  [Icons.mail_open]: <MUIDraftsIcon />,
  [Icons.mail]: <MUIMailIcon />,
  [Icons.menu]: <MUIMenuIcon />,
  [Icons.notifications]: <MUINotificationsIcon />,
  [Icons.open_new]: <MUIOpenInNew />,
  [Icons.pause]: <MUIPauseIcon />,
  [Icons.play]: <MUIPlayArrowIcon />,
  [Icons.power]: <MUIPowerSettingsNewIcon />,
  [Icons.print]: <MUIPrintIcon />,
  [Icons.refresh]: <MUIRefreshIcon />,
  [Icons.right]: <MUIChevronRightIcon />,
  [Icons.save]: <MUISaveIcon />,
  [Icons.search]: <MUISearchIcon />,
  [Icons.send]: <MUISendIcon />,
  [Icons.settings]: <MUISettingsIcon />,
  [Icons.up_arrow]: <MUIArrowUpwardIcon />,
  [Icons.up]: <MUIKeyboardArrowUpIcon />,
  [Icons.visibility]: <MUIVisibilityIcon />,
  [Icons.visibility_off]: <MUIVisibilityOffIcon />,
  [Icons.warn]: <MUIWarningIcon />,
  [Icons.wifi_off]: <MUIWifiOffOutlinedIcon />,
  [Icons.wifi_on]: <MUIWifiOutlinedIcon />,
};
