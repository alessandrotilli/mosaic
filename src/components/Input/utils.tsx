import React from "react";
import { InputAdornment as MUIInputAdornment } from "@material-ui/core";

import { IInputAdornment } from "../../types/Input";
import IconButton from "../IconButton";

export const getAdornment = (adornment?: IInputAdornment) => {
  if (!adornment) {
    return;
  }

  const { icon, onClick } = adornment;
  return (
    <MUIInputAdornment position="end">
      <IconButton disabled={!onClick} icon={icon} onClick={onClick!} size="small" />
    </MUIInputAdornment>
  );
};
