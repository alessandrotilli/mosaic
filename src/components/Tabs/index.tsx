import React, { FC } from "react";
import Box from "@mui/material/Box";
import MUITab from "@mui/material/Tab";
import MUITabs from "@mui/material/Tabs";

import { TabsType } from "../../types/Tabs";
import Typography from "../Typography";

export const DATA_CY_DEFAULT = "tabs";

const Tabs: FC<TabsType> = ({ dataCy = DATA_CY_DEFAULT, labelList, color = "primary" }) => {
  const [value, setValue] = React.useState(0);

  const handleChange = (event: React.SyntheticEvent, newValue: number) => {
    setValue(newValue);
  };
  return (
    <Box>
      <MUITabs data-cy={dataCy} value={value} onChange={handleChange} textColor={color} indicatorColor={color}>
        {labelList.map((i, index) => {
          return <MUITab label={i.label} wrapped={i.wrapped} key={index} />;
        })}
      </MUITabs>
      {labelList[value].children && (
        <div role="tabpanel" id={`simple-tabpanel-${value}`}>
          <Box sx={{ p: 3 }}>
            <Typography>{labelList[value].children}</Typography>
          </Box>
        </div>
      )}
    </Box>
  );
};

export default Tabs;
