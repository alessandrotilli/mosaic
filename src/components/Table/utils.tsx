import React, { forwardRef, Fragment } from "react";
import { Action as MTActionType, Column as MTColumnType, Options as MTOptionsType } from "material-table";
import { Icons, IconSize } from "../../types/Icon";
import { TableActionPosition, ITableAction, ITableColumn } from "../../types/Table";
import Button from "../Button";
import Icon from "../Icon";
import IconButton from "../IconButton";
import { getDataCyForSubComponent, suppressEvent } from "../../utils";

// For default props refer to Options section here https://material-table.com/#/docs/all-props
export const DEFAULT_TABLE_OPTIONS: MTOptionsType = {
  debounceInterval: 500,
  draggable: false,
  emptyRowsWhenPaging: false,
};

export const actionAdapter = (action: ITableAction): MTActionType<object> => {
  const {
    callback,
    label,
    disabled = false,
    hidden = false,
    icon = undefined,
    position = TableActionPosition.default,
  } = action;
  return {
    disabled,
    hidden,
    icon: `${label}$${icon}`,
    onClick: (event, data) => {
      if (event) {
        suppressEvent(event);
      }
      callback(data);
    },
    position,
  };
};

export const actionComponentAdapter = (props: any, dataCy: string) => {
  const { data, action } = props;
  const { disabled, hidden, icon, position, onClick }: MTActionType<object> = action;
  if (hidden) {
    return null;
  }

  const displaysButton = position === TableActionPosition.default;
  const labelAndIconName = icon as string;
  const label = labelAndIconName.split("$")[0];
  const iconName = labelAndIconName.split("$")[1] as Icons;
  return (
    <Fragment>
      {displaysButton && (
        <Fragment>
          <span>&nbsp;&nbsp;&nbsp;&nbsp;</span>
          <Button
            dataCy={getDataCyForSubComponent(dataCy, `action-${label}`)}
            disabled={disabled}
            icon={!iconName ? undefined : { name: iconName }}
            label={label}
            onClick={() => onClick(undefined, data)}
          />
        </Fragment>
      )}
      {!displaysButton && (
        <Fragment>
          <IconButton
            dataCy={getDataCyForSubComponent(dataCy, `action-${label}`)}
            disabled={disabled}
            icon={iconName}
            onClick={() => onClick(undefined, data)}
            size={IconSize.small}
          />
          <span>&nbsp;&nbsp;</span>
        </Fragment>
      )}
    </Fragment>
  );
};

export const columnAdapter = (column: ITableColumn): MTColumnType<object> => {
  const { label, path, render, width } = column;
  return {
    ...(!!width && ({ width } as object)),
    field: path,
    render,
    title: label,
  };
};

export const iconAdapter = (name: Icons, size = IconSize.default) =>
  forwardRef((props: any, ref: any) => {
    return <Icon forwarded={{ ...props, ref }} name={name} size={size} />;
  });
