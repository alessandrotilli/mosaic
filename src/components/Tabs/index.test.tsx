import renderer from "react-test-renderer";

import { TabsType } from "../../types/Tabs";
import { getTestableComponent, IPartialTestOptions, ITestOptions } from "../../utils/tests";

import Tabs, { DATA_CY_DEFAULT } from ".";

const DEFAULT_TEST_OPTIONS: ITestOptions<TabsType> = {
  dataCy: DATA_CY_DEFAULT,
  domNode: "tabs",
  localized: true,
  props: { labelList: [{ label: "PAGE" }] },
};

const getTabsTestable = (options?: IPartialTestOptions<TabsType>) =>
  getTestableComponent(Tabs, DEFAULT_TEST_OPTIONS, options);

describe.skip("Tabs test suite:", () => {
  it("color", () => {
    const { element } = getTabsTestable({ props: { color: "primary" } });

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("Page", () => {
    const { element } = getTabsTestable({ props: { labelList: [{ label: "PAGE1" }, { label: "PAGE2" }] } });

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("Wrapped", () => {
    const { element } = getTabsTestable({
      props: { labelList: [{ label: "PAGE1" }, { label: "PAGE2", wrapped: true }] },
    });

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("children", () => {
    const { element } = getTabsTestable({
      props: { labelList: [{ label: "PAGE1", content: "Children" }] },
    });

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });

  it("orientation", () => {
    const { element } = getTabsTestable({
      props: { labelList: [{ label: "PAGE1" }], orientation: "vertical" },
    });

    const snapshotWrapper = renderer.create(element).toJSON();
    expect(snapshotWrapper).toMatchSnapshot();
  });
});
