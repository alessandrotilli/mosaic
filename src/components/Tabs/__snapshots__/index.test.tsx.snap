// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs test suite: Page 1`] = `
<div
  className="MuiBox-root css-0"
>
  <div
    className="MuiTabs-root css-1ujnqem-MuiTabs-root"
    data-cy="tabs"
  >
    <div
      className="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
      onScroll={[Function]}
      style={
        {
          "marginBottom": -0,
          "overflow": "hidden",
        }
      }
    >
      <div
        aria-orientation={null}
        className="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
        onKeyDown={[Function]}
        role="tablist"
      >
        <button
          aria-selected={true}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          PAGE1
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          PAGE2
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <span
        className="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
        style={
          {
            "left": 0,
            "width": 0,
          }
        }
      />
    </div>
  </div>
  <div
    aria-labelledby="horizontal-tab-0"
    id="simple-tabpanel-0"
    role="tabpanel"
    style={
      {
        "padding": 8,
        "width": "100%",
      }
    }
  />
</div>
`;

exports[`Tabs test suite: Wrapped 1`] = `
<div
  className="MuiBox-root css-0"
>
  <div
    className="MuiTabs-root css-1ujnqem-MuiTabs-root"
    data-cy="tabs"
  >
    <div
      className="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
      onScroll={[Function]}
      style={
        {
          "marginBottom": -0,
          "overflow": "hidden",
        }
      }
    >
      <div
        aria-orientation={null}
        className="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
        onKeyDown={[Function]}
        role="tablist"
      >
        <button
          aria-selected={true}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          PAGE1
          <span
            className="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
            style={{}}
          />
        </button>
        <button
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-wrapped css-zx9imm-MuiButtonBase-root-MuiTab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          PAGE2
        </button>
      </div>
    </div>
  </div>
  <div
    aria-labelledby="horizontal-tab-0"
    id="simple-tabpanel-0"
    role="tabpanel"
    style={
      {
        "padding": 8,
        "width": "100%",
      }
    }
  />
</div>
`;

exports[`Tabs test suite: children 1`] = `
<div
  className="MuiBox-root css-0"
>
  <div
    className="MuiTabs-root css-1ujnqem-MuiTabs-root"
    data-cy="tabs"
  >
    <div
      className="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
      onScroll={[Function]}
      style={
        {
          "marginBottom": -0,
          "overflow": "hidden",
        }
      }
    >
      <div
        aria-orientation={null}
        className="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
        onKeyDown={[Function]}
        role="tablist"
      >
        <button
          aria-selected={true}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          PAGE1
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <span
        className="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
        style={
          {
            "left": 0,
            "width": 0,
          }
        }
      />
    </div>
  </div>
  <div
    aria-labelledby="horizontal-tab-0"
    id="simple-tabpanel-0"
    role="tabpanel"
    style={
      {
        "padding": 8,
        "width": "100%",
      }
    }
  >
    Children
  </div>
</div>
`;

exports[`Tabs test suite: color 1`] = `
<div
  className="MuiBox-root css-0"
>
  <div
    className="MuiTabs-root css-1ujnqem-MuiTabs-root"
    data-cy="tabs"
  >
    <div
      className="MuiTabs-scrollableX MuiTabs-hideScrollbar css-agdkjh-MuiTabs-scrollbarSize"
      style={
        {
          "height": 99,
          "overflow": "scroll",
          "position": "absolute",
          "top": -9999,
          "width": 99,
        }
      }
    />
    <div
      className="MuiTabs-scroller MuiTabs-hideScrollbar MuiTabs-scrollableX css-69z67c-MuiTabs-scroller"
      onScroll={[Function]}
      style={
        {
          "marginBottom": -0,
          "overflow": "hidden",
        }
      }
    >
      <div
        aria-orientation={null}
        className="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
        onKeyDown={[Function]}
        role="tablist"
      >
        <button
          aria-selected={true}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          PAGE
          <span
            className="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
            style={{}}
          />
        </button>
      </div>
    </div>
  </div>
  <div
    aria-labelledby="horizontal-tab-0"
    id="simple-tabpanel-0"
    role="tabpanel"
    style={
      {
        "padding": 8,
        "width": "100%",
      }
    }
  />
</div>
`;

exports[`Tabs test suite: orientation 1`] = `
<div
  className="MuiBox-root css-k008qs"
>
  <div
    className="MuiTabs-root MuiTabs-vertical css-1g4ac1o-MuiTabs-root"
    data-cy="tabs"
  >
    <div
      className="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
      onScroll={[Function]}
      style={
        {
          "marginRight": -0,
          "overflow": "hidden",
        }
      }
    >
      <div
        aria-orientation="vertical"
        className="MuiTabs-flexContainer MuiTabs-flexContainerVertical css-lfwcke-MuiTabs-flexContainer"
        onKeyDown={[Function]}
        role="tablist"
      >
        <button
          aria-selected={true}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          PAGE1
          <span
            className="MuiTabs-indicator css-10d9dml-MuiTabs-indicator"
            style={{}}
          />
        </button>
      </div>
    </div>
  </div>
  <div
    aria-labelledby="vertical-tab-0"
    id="simple-tabpanel-0"
    role="tabpanel"
    style={
      {
        "padding": 8,
        "width": "100%",
      }
    }
  />
</div>
`;
